cmake/Sanitizers.cmake

```diff
@@ -0,0 +1,70 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2013-2022 The Foundry Visionmongers Ltd
+
+function(enable_sanitizers project_name)
+
+    if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID MATCHES ".*Clang")
+        option(ENABLE_COVERAGE "Enable coverage reporting for gcc/clang" OFF)
+
+        if (ENABLE_COVERAGE)
+            target_compile_options(${project_name} PRIVATE --coverage -O0 -g)
+            target_link_libraries(${project_name} PRIVATE --coverage)
+        endif ()
+
+        set(SANITIZERS "")
```

Comment:
I believe you can get away with not needing this. The appends will create it if necessary, and dereferencing if not defined won't be an error.

Reply:
Removed in ffc5a180 then, after simplifying in 7ab8022d (as per https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/185#discussion_r786206935), moved out the content of that .cmake file in f5d89252